### PR TITLE
[BREAKING] Enforce error on hex combined with denomination

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

**Review by ekpyron — COMMENTED** [review]
Changelog entry is missing, otherwise looks good!

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -2267,11 +2267,9 @@ void TypeChecker::endVisit(ElementaryTypeNameExpression const& _expr)
 
 void TypeChecker::endVisit(Literal const& _literal)
 {
-	bool const v050 = m_scope->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050);
-
 	if (_literal.looksLikeAddress())
 	{
-		// Assign type here if it even looks like an address. This prevents double error in 050 mode for invalid address
+		// Assign type here if it even looks like an address. This prevents double error for invalid address
```
Comment (aj07) [inline libsolidity/analysis/TypeChecker.cpp]: can we make it plural ?

File: Changelog.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ Breaking Changes:
    ``sizeof``, ``supports``, ``typedef`` and ``unchecked``.
  * General: Remove assembly instruction aliases ``sha3`` and ``suicide``
  * General: C99-style scoping rules are enforced now. This was already the case in the experimental 0.5.0 mode.
+ * General: Disallow combining hex numbers with unit denominations (e.g. 0x1e wei).
```
Comment (ekpyron) [inline Changelog.md]: ``This was already the case in the experimental 0.5.0 mode.`` missing ;).

**Review by ekpyron — COMMENTED** [review]
Apart from the changelog entry and maybe @aj07's comment about improving the code comment, this still looks fine.

File: Changelog.md (hunk 2 above)
Comment (erak) [reply Changelog.md]: Does it make sense to create a section for items that would have this note? This could then be a subsection of ``Breaking changes``

Comment (ekpyron) [reply Changelog.md]: Don't we already have a list of all experimental 0.5.0 features somewhere? In any case - we did add such a notice to all the other changelog entries, so we should probably stay with it... we can think about reformatting the changelog before the release (doing it now would require every PR to be rebased, etc.)...

Comment (erak) [reply Changelog.md]: Good point. Let's keep this for now and perhaps reorganize them before the release.

File: libsolidity/analysis/TypeChecker.cpp [hunk 3]
```diff
@@ -2267,11 +2267,9 @@ void TypeChecker::endVisit(ElementaryTypeNameExpression const& _expr)
 
 void TypeChecker::endVisit(Literal const& _literal)
 {
-	bool const v050 = m_scope->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050);
-
 	if (_literal.looksLikeAddress())
 	{
-		// Assign type here if it even looks like an address. This prevents double error in 050 mode for invalid address
+		// Assign type here if it even looks like an address. This prevents double error for invalid addresses
```
Comment (ekpyron) [inline libsolidity/analysis/TypeChecker.cpp]: Well - if one wants to be *really* pedantic, then one should probably change to ``This prevents double errors for invalid addresses.`` - but I'm fine with it as it is, it's just a code comment after all...

**Review by ekpyron — DISMISSED** [review]

File: libsolidity/analysis/TypeChecker.cpp (hunk 3 above)
Comment (leonardoalt) [reply libsolidity/analysis/TypeChecker.cpp]: Fixed it and squashed.

**Review by ekpyron — DISMISSED** [review]

**Comment by axic** [comment]
Looks good. Perhaps we should rebase it due to other things merged.

**Comment by leonardoalt** [comment]
Rebased

**Review by ekpyron — DISMISSED** [review]

File: Changelog.md [hunk 4]
```diff
@@ -25,6 +25,7 @@ Breaking Changes:
    ``sizeof``, ``supports``, ``typedef`` and ``unchecked``.
  * General: Remove assembly instruction aliases ``sha3`` and ``suicide``
  * General: C99-style scoping rules are enforced now. This was already the case in the experimental 0.5.0 mode.
+ * General: Disallow combining hex numbers with unit denominations (e.g. 0x1e wei). This was already the case in the experimental 0.5.0 mode.
```
Comment (chriseth) [inline Changelog.md]: Please add
```
``
```

**Pull request merged**